packages/react-shortcuts/README.md

```diff
@@ -0,0 +1,156 @@
+# `@shopify/react-shortcuts`
+
+![bundle size badge](https://img.shields.io/bundlephobia/minzip/@shopify/react-shortcuts.svg)
+
+Declaratively and efficiently match shortcut combinations in your react application.
+
+## Installation
+
+```bash
+$ yarn add @shopify/react-shortcuts
+```
+
+## Usage
+
+The library exposes three main parts, `<ShortcutProvider />`, `<Shortcut />` and `ShortcutManager`.
+
+### ShortcutProvider
+
+Wrapping your application in a `<ShortcutProvider />` allows you to use `<Shortcut />` components anywhere in your application, internally sharing a single `ShortcutManager` instance to minimize listeners and collisions.
+
+```ts
+// App.ts
+
+import React from 'react';
+import {ShortcutProvider} from '@shopify/react-shortcuts';
+
+export default function App() {
+  <ShortcutProvider>{/* the rest of your app */}</ShortcutProvider>;
+}
+```
+
+### Shortcut
+
+Shortcut is used to register a new keyboard shortcut to `ShortcutManager`. It can be triggered globally or only when a node is focused.
+
+**Note: a `<Shortcut />` must have a `<ShortcutProvider />` somewhere above it in the tree.**
+
+#### API
+
+```ts
+export interface Props {
+  /*
+    keys that, when pressed sequentially, will trigger `onMatch`
+  */
+  keys: Key[];
+  /*
+    a callback that will trigger when the key combination is pressed
+  */
+  onMatch(keys: Key[]): void;
+  /*
+    a node that, when supplied, will be used to only fire `onMatch` when it is focused
+  */
+  node?: HTMLElement | null;
+  /*
+    a boolean that lets you temporarily disable the shortcut
+  */
+  ignoreInput?: boolean;
+  /*
+    a boolean that lets you opt out of swalloing the key event and let it propagate
+  */
+  allowDefault?: boolean;
+}
+```
+
+#### Basic example
```

Comment:
Can we also add an example for modifier keys? E.g. `<Shortcut key={['ctrlKey', 't', 'o']} onMatch={() => console.log('foo')} />`, and mention that internally, we are checking the `event.key`?

Also will need to write a matching test for it, and update the `ModifierKeys` enum to contain the updated key names.